benchmark/scenarios/_request_load.py

```diff
@@ -128,10 +128,19 @@ def handle_request_error(result):
             write_failure(result)
 
         for i in range(self.request_rate):
+            t0 = self.reactor.seconds()
+
             d = self.scenario_setup.make_request()
+
+            def get_time(_ignore):
+                return self.reactor.seconds() - t0
+            d.addCallback(get_time)
+
             self.rate_measurer.request_sent()
-            d.addCallbacks(self.rate_measurer.response_received,
-                           errback=handle_request_error)
+            d.addCallbacks(
+                self.rate_measurer.response_received,
+                handle_request_error
+            )
```

Comment:
Initially I thought it was a bug that escaped because the result gets wrapped in `maybeDeferred`, so it doesn't cause any detectable errors, except that it might mean that the function is re-entered while it is still running.  But even that isn't actually a problem.  It just makes it less likely that a call will be skipped, and have to be caught up using the `count` parameter.

In fact, gathering the Deferreds would possibly break things because we're trying to launch `request_rate` operations per second, and waiting for a laggard would drop the rate too far.

Essentially, the code is correct to return immediately since we are trying to start a certain number of requests per second, and we don't want this function to wait for anything.  The Deferreds that are created inside the function are handled correctly by reporting to the Rate Measurer. And we have a separate check for overload in `check_rate`.

I do see 2 things that need to be fixed:
1. We should return `succeed(None)` and add some comments to indicate that we know what we're doing.
2. There is a bug in that if the loop is delayed, the rate measurer catches up (it gets called `count` times), but we only start one batch of requests, not `count` batches.  However, I will fix these in a separate PR.